src/go/rpk/pkg/cli/cmd/debug/bundle.go

```diff
@@ -0,0 +1,552 @@
+ - DNS: The DNS info as reported by 'dig', using the hosts in
+  /etc/resolv.conf.
+ - Virtual memory stats: As reported by 'vmstat'.
+ - Socket info: The active sockets data output by 'ss'.
+ - Running process info: As reported by 'top'.
+ - Data directory structure: The 'tree' output describing the data directory
+  contents.
+`,
+		SilenceUsage: true,
+		RunE: func(ccmd *cobra.Command, args []string) error {
+			mgr := config.NewManager(fs)
+			conf, err := mgr.ReadOrFind(configFile)
+			if err != nil {
+				return err
+			}
+			// Since we want to use and bundle in the actual config,
+			// common.FindConfigFile can't be used (it defaults to returning
+			// config.Default()). Therefore, we check that the config is in a
+			// the path given by --config or a default path as defined in
+			// Manager#ReadOrFind.
+			configClosure := func() (*config.Config, error) {
+				return conf, nil
+			}
+			brokersClosure := common.DeduceBrokers(
+				common.CreateDockerClient,
+				configClosure,
+				&brokers,
+			)
+			tlsClosure := common.BuildKafkaTLSConfig(fs, &enableTLS, &certFile, &keyFile, &truststoreFile, configClosure)
+			kAuthClosure := common.KafkaAuthConfig(&user, &password, &mechanism, configClosure)
+			admin := common.CreateAdmin(brokersClosure, configClosure, tlsClosure, kAuthClosure)
+
+			adm, err := admin()
+			if err != nil {
+				log.Error("Couldn't initialize API admin")
+				return err
+			}
+			defer adm.Close()
+
+			return executeSnapshot(fs, conf, adm, logsSince, logsUntil, timeout)
+		},
+	}
+	command.Flags().DurationVar(
+		&timeout,
+		"timeout",
+		10*time.Second,
+		"The maximum amount of time to wait for child commands (ss, ifconfig, "+
+			"dig, etc.) to be executed. "+
+			"The value passed is a sequence of decimal numbers, each with optional "+
+			"fraction and a unit suffix, such as '300ms', '1.5s' or '2h45m'. "+
+			"Valid time units are 'ns', 'us' (or 'µs'), 'ms', 's', 'm', 'h'",
+	)
+	command.Flags().StringVar(
+		&logsSince,
+		"logs-since",
+		"",
+		fmt.Sprintf(`Include log entries on or newer than the specified date. %s`, timeHelpText),
+	)
+	command.Flags().StringVar(
+		&logsUntil,
+		"logs-until",
+		"",
+		fmt.Sprintf(`Include log entries on or older than the specified date. %s`, timeHelpText),
+	)
+
+	common.AddKafkaFlags(
+		command,
+		&configFile,
+		&user,
+		&password,
+		&mechanism,
+		&enableTLS,
+		&certFile,
+		&keyFile,
+		&truststoreFile,
+		&brokers,
+	)
+	return command
+}
+
+func executeSnapshot(
+	fs afero.Fs,
+	conf *config.Config,
+	admin sarama.ClusterAdmin,
+	logsSince, logsUntil string,
+	timeout time.Duration,
+) error {
+	mode := os.FileMode(0755)
+	timestamp := time.Now().Unix()
+	filename := fmt.Sprintf("%d-bundle.zip", timestamp)
+	f, err := fs.OpenFile(
+		filename,
+		os.O_CREATE|os.O_WRONLY,
+		mode,
+	)
+	if err != nil {
+		return fmt.Errorf("couldn't create bundle file: %w", err)
+	}
+	defer f.Close()
+
+	grp := multierror.Group{}
+
+	m := &sync.Mutex{}
```

Comment:
This can be left out, especially if switching the field in `stepParams` from a pointer to a non-pointer

Reply:
This can be left out.